selfdrive/ui/sound.hpp

```diff
@@ -1,17 +1,28 @@
-#ifndef __SOUND_HPP
-#define __SOUND_HPP
-
+#pragma once
+#include <map>
 #include "cereal/gen/cpp/log.capnp.h"
-
 typedef cereal::CarControl::HUDControl::AudibleAlert AudibleAlert;
-
-void ui_sound_init();
-void ui_sound_destroy();
-
-void set_volume(int volume);
-
-void play_alert_sound(AudibleAlert alert);
-void stop_alert_sound(AudibleAlert alert);
-
+#if defined(QCOM) || defined(QCOM2)
+#include <SLES/OpenSLES.h>
+#include <SLES/OpenSLES_Android.h>
 #endif
-
+class Sound {
+ public:
+  Sound() = default;
+  bool init(int volumn);
+  bool play(AudibleAlert alert, int repeat = 0);
+  bool stop();
+  void setVolume(int volume, int timeout_seconds = 0);
+  inline AudibleAlert currentSound() const { return currentSound_; }
+  ~Sound();
+#if defined(QCOM) || defined(QCOM2)
+ private:
+  SLObjectItf engine_ = NULL;
+  SLObjectItf outputMix_ = NULL;
+  int repeat_ = 0, last_volume_ = 0;
+  double last_set_volume_time_ = 0;
+  struct Player;
+  std::map<AudibleAlert, Player*> player_;
+#endif
+  AudibleAlert currentSound_ = AudibleAlert::NONE;
```

Comment:
This can be either public or private depending on `defined(QCOM) || defined(QCOM2)`